### PR TITLE
Add types for cytoscape-dagre

### PR DESCRIPTION
**Pull request by DieserFelix**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by janniclas — APPROVED** [review]
I reviewed the changes by comparing it to the official documentation found [here](https://github.com/cytoscape/cytoscape.js-dagre). The definition matches and the tests all execute. Good to go from my point of view.

File: types/cytoscape-dagre/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
```
Comment (peterblazejewicz) [inline types/cytoscape-dagre/tsconfig.json:6]: just question, why you've added `dom` to the `lib`? (not obvious from the code)
thx!

**Comment by peterblazejewicz** [comment]
I think you will either have to add `export as namespace cytoscapeDagre` or add augmentation for `global` object, as this can be used directly outside of the modules on a web page. Is that module exported that way, just like `cytoscope`?

**Comment by DieserFelix** [comment]
> I think you will either have to add `export as namespace cytoscapeDagre` or add augmentation for `global` object, as this can be used directly outside of the modules on a web page. Is that module exported that way, just like `cytoscope`?

I'm not sure I understand the question. Adding `export as namespace cytoscapeDagre` would not break anything so I am not against it, but I don't understand why it's necessary. Can you elaborate?

File: types/cytoscape-dagre/tsconfig.json (hunk 1 above)
Comment (peterblazejewicz) [reply types/cytoscape-dagre/tsconfig.json:6]: thx!

**Comment by peterblazejewicz** [comment]
My question is, are there any users expected to be using this package as global script? (no module, plain JavaScript). Just like in `examples` directroy under `gh-pages`:
https://github.com/cytoscape/cytoscape.js-dagre/blob/gh-pages/index.html

**Comment by DieserFelix** [comment]
> My question is, are there any users expected to be using this package as global script? (no module, plain JavaScript). Just like in `examples` directory under `gh-pages`:
> https://github.com/cytoscape/cytoscape.js-dagre/blob/gh-pages/index.html
> https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library

In general, it is possible to use the `cytoscape-dagre` extension as a global script in plain JavaScript, analog to `cytoscape`.

Our understanding of namespaces isn't that great to be honest, but if it's the intended way we are fine with adding the namespace export (and have added it in the latest commit).

File: types/cytoscape-dagre/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// Type definitions for cytoscape-dagre 2.3
+// Project: https://github.com/cytoscape/cytoscape.js-dagre
+// Definitions by: Felix Barczewicz <https://github.com/DieserFelix>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
```
Comment (peterblazejewicz) [inline types/cytoscape-dagre/index.d.ts:5]:
```suggestion
import cytoscape = require('cytoscape');

```
I believe you need this, it just works now, as there is direct import in test file

**Comment by peterblazejewicz** [comment]
> Our understanding of namespaces isn't that great to be honest, but if it's the intended way we are fine with adding the namespace export (and have added it in the latest commit).

Thanks! Yes, it's just a convention in TS, and only for UMD modules, has no other meaning and is not a regular namespace:
https://github.com/Microsoft/TypeScript/issues/7125

File: types/cytoscape-dagre/index.d.ts (hunk 2 above)
Comment (DieserFelix) [reply types/cytoscape-dagre/index.d.ts:5]: Oh, you are absolutely correct. Added it in the latest commit. Thank you!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by DieserFelix** [comment]
Ready to merge

**Pull request merged**